### DISCUSSION TITLE
Amending entries in the commands_en.txt

### DISCUSSION THREAD

**Discussion by ghost** [toverainc/willow#115] [Q&A] [answered]
If I select the Multinet in config and compile .  the commands_ent.txt  file is created and all my lights and switches are created . How can I edit this file and  build a new flash file without the file being repopulated and stop my edits from being overwritten

**Comment by kristiankielhofner** [comment, marked as answer]
Willow is early, and local command recognition is even earlier!

One note - DO NOT edit commands_en.txt on it's own. There are two issues:

1) Home Assistant returns entities randomly. They can get out of order so that the commands defined in commands_en.txt don't line up with the index we use for what is sent to Home Assistant. It will start doing all kinds of strange things.

2) Without some programmatic validation of entries on the frontend you can crash the device. We do this in our current generation process but if you edit them manually you're kind of on your own...

We currently have two measures planned for this:

1) Allow editing of commands during the current configure, build, and flash process. This is relatively simple but we still need to figure out a user interface that can also do sanity checking of the user defined grammar as there are inherent limitations with on device local command support.

2) Fully dynamic configuration of Willow itself. This is a much bigger task (as I'm sure you can imagine) but we will include the ability to do user configured updates and/or refreshing based on changes in Home Assistant and other supported platforms.

I don't currently have an ETA on either of these, it seems many people have received their ESP BOXes and we're working through a variety of issues ATM.

**Reply by tombloemker** [reply]
This may be off topic...
My commands_ent.txt file has 22 entries, only one worked initially and I found out that device 'Jills Lamp' is not 'Jill's Lamp'. (with an apostrophe) I assume I must rename all of my entities to match Willow's output? 

Disregard. :( Alias names work just fine. Even dimmers are working! So awesome!

**Comment by ghost** [comment]
Thanks for stunningly quick response  :)

**Reply by kristiankielhofner** [reply]
I try!
